arm-insights/2015-04-01/swagger/insightsClient.json

```diff
@@ -0,0 +1,740 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "InsightsClient",
+    "version": "2015-04-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/providers/microsoft.insights/eventtypes/management/eventcategories": {
+      "get": {
+        "tags": [
+          "EventCategories"
+        ],
+        "operationId": "EventCategories_List",
+        "description": "The list of event categories.",
+        "parameters": [
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "default": "2015-04-01",
+            "description": "The api-version to send to the server"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to get the list of event categories",
+            "schema": {
+              "$ref": "#/definitions/EventCategoryCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/microsoft.insights/eventtypes/management/values": {
+      "get": {
+        "tags": [
+          "Events"
+        ],
+        "operationId": "Events_List",
+        "description": "The List Event Values operation lists the events.",
+        "parameters": [
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "default": "2015-04-01",
+            "description": "The api-version to send to the server"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "The filter to apply on the operation."
+          },
+          {
+            "name": "$select",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The list of property names to be returned. Valid values are: authorization, channels, claims, correlationId, description, eventDataId, eventName, eventTimestamp, httpRequest, level, operationId, operationName, properties, resourceGroupName, resourceProviderName, resourceId, status, submissionTimestamp, subStatus, subscriptionId"
```

Comment:
The EventData model indicates that "channels", "level", "eventTimestamp" and "submissionTimestamp" are all required. What happens if I don't include one or more of those fields in the $select clause?